File: 卡码/代码随想录/1/08-贪心算法/C++/02-中等/LeetCode_0376.cpp
//376. 摆动序列
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
	int wiggleMaxLength(vector<int>& nums) {
		if(nums.size()<=1) return nums.size();
		int res=1,pre=0,cur=0;
		for(int i=1;i<nums.size();i++){
			cur=nums[i]-nums[i-1];
			if(pre>=0&&cur<0||pre<=0&&cur>0){
				res++;
				pre=cur;
			}
		}
		return res;
	}
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/1/08-贪心算法/Js/02-中等/LeetCode_0376.js
//376. 摆动序列
const wiggleMaxLength = function(nums) {
  if(nums.length<=1) return nums.length;
  let res=1,pre=0,cur=0;
  for(let i=1;i<nums.length;i++){
    cur=nums[i]-nums[i-1];
    if(pre<=0&&cur>0||pre>=0&&cur<0){
      res++;
      pre=cur;
    }
  }
  return res;
};